apps/web/src/components/buttons/create-organization-button.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+
+import { useDisclosure } from '@mantine/hooks';
+import { ActionIcon, Button, Flex, Group, Modal, Text, TextInput, Tooltip } from '@mantine/core';
+import { IconPlus } from '@tabler/icons-react';
+import React, { useState } from 'react';
+import { type ReactNode } from 'react';
+import { useTranslations } from 'next-intl';
+import { useForm } from '@mantine/form';
+import { logger } from '@repo/logger';
+import { type CreateOrganizationMutationVariables } from '@/graphql/generated/schema-server';
+import { createOrganization, switchOrganization } from '@/app/(authenticated)/organization/actions';
+import { changeJWT } from '@/app/(unauthenticated)/actions';
+
+export default function CreateOrganizationButton(): ReactNode {
+  const t = useTranslations('organization');
+  const [opened, { open, close }] = useDisclosure(false);
+  const [isPending, setIsPending] = useState<boolean>(false);
+  const form = useForm({
+    mode: 'controlled',
+    initialValues: {
+      name: '',
+    },
+  });
+
+  const handleSubmit = (values: CreateOrganizationMutationVariables): void => {
+    setIsPending(true);
+    void createOrganization(values)
+      .then((res) => {
+        if (!res.success) {
+          logger.error(res.error);
+          return new Error(res.error);
+        }
+
+        // On success, switch to the newly created organization!
+        void switchOrganization({ organizationId: res.data.createOrganization.id })
+          .then((switchRes) => {
+            if (switchRes.success) {
+              void changeJWT(
+                switchRes.data.switchOrganization.token,
+                switchRes.data.switchOrganization.refreshToken,
+              ).then(() => {
+                window.location.reload();
+              });
+            }
+          })
+          .catch((error: unknown) => {
+            logger.error(error);
+          });
+      })
+      .catch((error: unknown) => {
+        logger.error(error);
+      })
+      .finally(() => {
+        setIsPending(false);
+        form.reset();
+      });
```

Comment:
```suggestion
    void createOrganization(values).then((res) => {
      setIsPending(false);
      form.reset();
      if (isAError(res)) {
        logger.error(res.message);
        return new Error(res.message);
      }
      window.location.reload();
    });
```
I think it makes more sense to move these steps on the server side, for example the `createOrganization` could end up being:
```
export async function createOrganization(values: CreateOrganizationMutationVariables): Promise<AError | undefined> {
  const createOrgResp = await handleUrqlRequest(urqlClientSdk().createOrganization(values));
  if (!createOrgResp.success) {
    return new AError(createOrgResp.error);
  }
  const switchOrgResp = await switchOrganization({ organizationId: createOrgResp.data.createOrganization.id });
  if (!switchOrgResp.success) {
    return new AError(switchOrgResp.error);
  }
  await changeJWT(switchOrgResp.data.switchOrganization.token, switchOrgResp.data.switchOrganization.refreshToken);
  return undefined;
}
```

Reply:
Now that both are on the server side, upon creating an organization and changing the JWT token to the new token, we get the following error regarding JWS. 
This seems to sign out the user upon refresh, even if the new token has been set correctly in the cookies.
@trixobird 

```
web:dev:  [JWSSignatureVerificationFailed: signature verification failed] {
web:dev:   code: 'ERR_JWS_SIGNATURE_VERIFICATION_FAILED',
web:dev:   name: 'JWSSignatureVerificationFailed',
web:dev:   message: 'signature verification failed'
web:dev: }
```